app/services/waste_carriers_engine/renewal_completion_service.rb

```diff
@@ -36,8 +41,24 @@ def complete_renewal
 
     private
 
-    def find_original_registration
-      Registration.where(reg_identifier: transient_registration.reg_identifier).first
+    def registration
+      @registration ||= Registration.where(reg_identifier: transient_registration.reg_identifier).first
+    end
+
+    def raise_completion_error
+      # TODO: Temporaty debugging code for issue https://eaflood.atlassian.net/browse/RUBY-885
+      message = "Registration number: #{transient_registration.reg_identifier}"
+
+      raise(WrongWorkflowState, message) unless transient_registration.renewal_application_submitted?
+      raise(StillUnpaidBalance, message) if transient_registration.unpaid_balance?
+      raise(PendingConvictionCheck, message) if transient_registration.pending_manual_conviction_check?
+
+      unless %w[ACTIVE EXPIRED].include?(registration.metaData.status)
+        message = "#{message}. Status: #{registration.metaData.status}"
+        raise(WrongStatus, message)
+      end
+
+      raise(CannotComplete, message)
```

Comment:
There is no test for this, as this is "In case is not something else". Hence, if raised, we have to start from square 0 and search for a different cause.